test/lib/ansible_test/_internal/ansible_util.py

```diff
@@ -123,14 +123,16 @@ def check_pyyaml(args, version):
     :type args: EnvironmentConfig
     :type version: str
     """
-    if version in CHECK_YAML_VERSIONS:
-        return
+    try:
+        return CHECK_YAML_VERSIONS[version]
+    except KeyError:
+        pass
 
     python = find_python(version)
     stdout, _dummy = run_command(args, [python, os.path.join(ANSIBLE_TEST_DATA_ROOT, 'yamlcheck.py')], capture=True)
 
     if args.explain:
-        return
+        return {'yaml': False, 'cloader': False}
```

Comment:
@mattclay is this correct?  From the way the code is structured, this seemed like a reasonable value to put here put I don't know what explain is so if you have a better suggestion.....

Reply:
Explain mode (`--explain`) tells ansible-test to show what it would do without actually running the tests. Usually that means not running external commands, which is how `check_pyyaml` is currently working.

Now that `check_pyyaml` is being used to control program flow and not just display a warning, we should change it to run `yamlcheck.py` even in explain mode, rather than returning hard-coded results.  To do that, add `always=True` to this function call:

https://github.com/ansible/ansible/blob/7767f0bb2381bc8980be6f512937d5415c635a28/test/lib/ansible_test/_internal/ansible_util.py#L130

Then you can remove the explain conditional here:

https://github.com/ansible/ansible/blob/7767f0bb2381bc8980be6f512937d5415c635a28/test/lib/ansible_test/_internal/ansible_util.py#L132-L133

That way running with `--explain` (or `-e`) will provide the correct results.

Reply:
Thanks.  Changed.